### PR TITLE
feat(ui): align frontend based on creation flows designs

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-991

### CODE REVIEW

File: libs/shared/console-shared/src/lib/job-configure-settings/ui/job-configure-settings.tsx [hunk 1]
```diff
@@ -42,47 +48,41 @@ export function JobConfigureSettings(props: JobConfigureSettingsProps) {
...
-            <ExternalLink href="https://crontab.guru" size="xs">
-              CRON expression builder
-            </ExternalLink>
-          </div>
-          <TimezoneSetting className="mb-3" />
-          <EntrypointCmdInputs />
-        </>
+          <TimezoneSetting />
+          {/* XXX: This should be migrated to general settings page */}
+          {props.legacyMode && <EntrypointCmdInputs />}
```
Comment (RemiBonnet) [inline libs/shared/console-shared/src/lib/job-configure-settings/ui/job-configure-settings.tsx:64]: Can you create a ticket for this ? I'm afraid we'll forget it

File: libs/shared/ui/src/lib/components/inputs/input-select/input-select.tsx [hunk 2]
```diff
@@ -20,6 +20,7 @@ export interface InputSelectProps {
   value?: string | string[]
   options: Value[]
   disabled?: boolean
+  hint?: ReactNode
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/inputs/input-select/input-select.tsx:23]:
```suggestion
  hint?: ReactNode | string
```
?

File: libs/shared/ui/src/lib/components/inputs/input-text-area/input-text-area.tsx [hunk 3]
```diff
@@ -7,12 +7,13 @@ export interface InputTextAreaProps {
   onChange?: (e: FormEvent<HTMLTextAreaElement>) => void
   className?: string
   disabled?: boolean
+  hint?: ReactNode
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/inputs/input-text-area/input-text-area.tsx:10]:
```suggestion
  hint?: ReactNode | string
```
?

File: libs/shared/ui/src/lib/components/inputs/input-text/input-text.tsx [hunk 4]
```diff
@@ -8,6 +8,7 @@ export interface InputTextProps {
   type?: 'text' | 'number' | 'password' | 'email' | 'date' | 'datetime' | 'time'
   className?: string
   onChange?: ChangeEventHandler<HTMLInputElement>
+  hint?: ReactNode
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/inputs/input-text/input-text.tsx:11]:
```suggestion
  hint?: ReactNode | string
```
?

File: libs/shared/ui/src/lib/components/funnel-flow/funnel-flow.tsx [hunk 5]
```diff
@@ -21,17 +21,17 @@ const FunnelFlowContent = (props: FunnelFlowProps) => {
...
             <h4 className="text-neutral-400 text-sm font-medium">{props.currentTitle}</h4>
           </div>
         </div>
         {props.onExit && (
           <div className="border-l border-l-neutral-200 pl-5 h-full flex items-center">
-            <ButtonLegacy onClick={props.onExit} style={ButtonLegacyStyle.STROKED} className="btn--no-min-w">
-              Close <Icon iconName="xmark" className="ml-2" />
-            </ButtonLegacy>
+            <Button onClick={props.onExit} variant="surface" size="md">
+              Close <Icon iconName="xmark" iconStyle="regular" className="text-xl ml-2" />
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/funnel-flow/funnel-flow.tsx:33]:
```suggestion
              Close <Icon iconName="xmark" iconStyle="regular" className="text-base ml-2" />
```

File: libs/pages/services/src/lib/ui/page-job-create/step-general/step-general.tsx [hunk 6]
```diff
@@ -26,84 +18,61 @@ export interface StepGeneralProps {
...
-              onChange={field.onChange}
-              value={field.value}
-              label="Application name"
-              error={error?.message}
-            />
-          )}
-        />
+        <Section className="gap-4">
+          <Heading>General</Heading>
+          <GeneralSetting label="Application name" />
```
Comment (RemiBonnet) [inline libs/pages/services/src/lib/ui/page-job-create/step-general/step-general.tsx:40]:
```suggestion
          <GeneralSetting label="Job name" />
```

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks @ctjhoa ! Below some feedback: 

### All (except Job)
- Missing separator after `Description` 
Screenshot 2024-04-03 at 10 17 25
Screenshot 2024-04-03 at 10 17 36

### Database
- Missing margin for Callout
Screenshot 2024-04-03 at 10 29 20

### Helm
- `Back` buttons are the old version for all steps (except summary)

File: libs/shared/ui/src/lib/components/inputs/input-select/input-select.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/inputs/input-select/input-select.tsx:23]: `ReactNode` already includes `string`
```ts
    type ReactNode =
        | ReactElement
        | string
        | number
        | Iterable<ReactNode>
        | ReactPortal
        | boolean
        | null
        | undefined
        | DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES[
            keyof DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES
        ];
```

File: libs/shared/ui/src/lib/components/inputs/input-text-area/input-text-area.tsx (hunk 3 above)
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/inputs/input-text-area/input-text-area.tsx:10]: `ReactNode` already includes `string`
```ts
    type ReactNode =
        | ReactElement
        | string
        | number
        | Iterable<ReactNode>
        | ReactPortal
        | boolean
        | null
        | undefined
        | DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES[
            keyof DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES
        ];
```

File: libs/shared/ui/src/lib/components/inputs/input-text/input-text.tsx (hunk 4 above)
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/inputs/input-text/input-text.tsx:11]: `ReactNode` already includes `string`
```ts
    type ReactNode =
        | ReactElement
        | string
        | number
        | Iterable<ReactNode>
        | ReactPortal
        | boolean
        | null
        | undefined
        | DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES[
            keyof DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES
        ];
```

File: libs/pages/services/src/lib/ui/page-job-create/step-general/step-general.tsx (hunk 6 above)
Comment (ctjhoa) [reply libs/pages/services/src/lib/ui/page-job-create/step-general/step-general.tsx]: Fix https://github.com/Qovery/console/pull/1258/commits/5a8fe7e64df3a3805e41d639f47212dda938f979

File: libs/shared/ui/src/lib/components/funnel-flow/funnel-flow.tsx (hunk 5 above)
Comment (ctjhoa) [reply libs/shared/ui/src/lib/components/funnel-flow/funnel-flow.tsx]: Fix https://github.com/Qovery/console/pull/1258/commits/5a8fe7e64df3a3805e41d639f47212dda938f979

File: libs/shared/console-shared/src/lib/job-configure-settings/ui/job-configure-settings.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/shared/console-shared/src/lib/job-configure-settings/ui/job-configure-settings.tsx:64]: Done https://qovery.atlassian.net/browse/FRT-1073

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**